### PR TITLE
Add getters to fetch modifier info during recipe creation

### PR DESCRIPTION
**Pull request by rahul-tuli**
This PR adds getters for fetching `ModifierBuildInfo` objects from registry based off of args specified by user

TODO:
- [ ] Change base to `sparseml.recipe_template.dev` before landing

### CODE REVIEW

File: src/sparseml/recipe_template/utils.py [hunk 1]
```diff
@@ -142,3 +150,110 @@ def modifier_recipe_variables(self) -> Dict[str, Any]:
...
     ),
 }
+
+
+# getters
+
+
+def get_quantization_info(
+    quantization: Union[str, bool] = False,
+    target: str = "vnni",
```
Comment (dbogunowicz) [inline src/sparseml/recipe_template/utils.py:160]: shall we add checks to make sure that the argument for `target` is one of the [vnni, tensorrt etc]?

File: src/sparseml/recipe_template/utils.py [hunk 2]
```diff
@@ -142,3 +150,110 @@ def modifier_recipe_variables(self) -> Dict[str, Any]:
...
+    if isinstance(
+        modifier_info.modifier, (MagnitudePruningModifier, GMPruningModifier)
+    ):
+        modifier_info.update(
+            updated_fields={
+                "mask_type": mask_type,
+                "global_sparsity": global_sparsity,
+            }
+        )
+    elif isinstance(modifier_info.modifier, ACDCPruningModifier):
```
Comment (dbogunowicz) [inline src/sparseml/recipe_template/utils.py:219]: acdc also has an argument `mask_type`, why is it missing here?

File: src/sparseml/recipe_template/utils.py [hunk 3]
```diff
@@ -12,20 +12,28 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Any, Dict, Optional
```
Comment (dbogunowicz) [inline src/sparseml/recipe_template/utils.py:15]: I'd love to see some unit testing for this batch of four PRs. Both for hardening and documentation purposes.

**Comment by rahul-tuli** [comment]
Closing this one out due to a design change that does not need this code

**Pull request closed without merging**